### PR TITLE
OBJ-225 New end-point to get a company's action code

### PR DESCRIPTION
**Pull request by mwestacott**

### CODE REVIEW

File: specs/EmergencyAuthCodeEligibleOfficers.yml [hunk 1]
```diff
@@ -3,12 +3,15 @@ info:
   title: Companies House Internal Oracle Query API
   version: '1.0'
 servers:
-  - url: http://localhost:18XXX
+  - url: http://localhost:18565
```
Comment (mwestacott) [inline specs/EmergencyAuthCodeEligibleOfficers.yml:6]: Swagger doesn't like alpha characters in the port number - it messes up the test CURL statements it generates. Changed this to be the port used when service runs in Vagrant dev env.

File: specs/EmergencyAuthCodeEligibleOfficers.yml [hunk 2]
```diff
@@ -3,12 +3,15 @@ info:
...
 tags:
   - name: emergency-auth-code-officers
```
Comment (ICoxCH) [inline specs/EmergencyAuthCodeEligibleOfficers.yml:8]: This file is called EmergencyAuthCodeEligibleOfficers, it doesn't seem right to add our endpoint to this spec.
Can we just create a seperate yaml file for our query?
Something like specs/CompanyActionCode.yml?

Comment (mwestacott) [reply specs/EmergencyAuthCodeEligibleOfficers.yml:8]: Oh yes, was going to check with someone about either renaming this file to something more generic or have our endpoint separate. Will do that.

Comment (mwestacott) [reply specs/EmergencyAuthCodeEligibleOfficers.yml:8]: Spoke to @jphillips2121 . Will rename the file to make it more generic.

File: src/main/java/uk/gov/ch/corporatebody/controller/CorporateBodyController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+@RestController
+public class CorporateBodyController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OracleQueryApplication.APPLICATION_NAME_SPACE);
+
+    @Autowired
+    private CorporateBodyService corporateBodyService;
+
+
```
Comment (markpit) [inline src/main/java/uk/gov/ch/corporatebody/controller/CorporateBodyController.java:24]: double blank lines

File: src/main/java/uk/gov/ch/corporatebody/controller/CorporateBodyController.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    private CorporateBodyService corporateBodyService;
+
+    
+    @GetMapping("/company/{companyNumber}/action-code")
+    public ResponseEntity getActionCode(@PathVariable String companyNumber) {
+
+        LOGGER.info("Calling service to retrieve action code for company number " + companyNumber);
+
+        try {
+            int actionCode = corporateBodyService.getActionCode(companyNumber);
```
Comment (markpit) [inline src/main/java/uk/gov/ch/corporatebody/controller/CorporateBodyController.java:31]: can you make this a Long? i think an Oracle number can be greater in size than in int can handle

File: src/main/java/uk/gov/ch/corporatebody/dao/CorporateBodyDao.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    private static final String GET_ACTION_CODE_SQL = "SELECT action_code_type_id FROM corporate_body WHERE incorporation_number = ?";
+    
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    
+    public int getActionCode(String companyNumber) throws CorporateBodyNotFoundException {
+        try {
+            int actionCode = jdbcTemplate.queryForObject(GET_ACTION_CODE_SQL, Integer.class, companyNumber);
```
Comment (markpit) [inline src/main/java/uk/gov/ch/corporatebody/dao/CorporateBodyDao.java:26]: see other comment, Oracle Number can be > max int size

File: src/main/java/uk/gov/ch/corporatebody/controller/CorporateBodyController.java (hunk 3 above)
Comment (mwestacott) [reply src/main/java/uk/gov/ch/corporatebody/controller/CorporateBodyController.java:24]: Will remove.

**Review by markpit — APPROVED** [review]

**Review by ICoxCH — APPROVED** [review]

**Pull request merged**